CRM/Sqltasks/Task.php

```diff
@@ -188,15 +189,46 @@ public function setConfiguration($config, $writeTrough = FALSE) {
    * Append log messages
    *
    * @param $message
+   * @param null $type
+   * @param bool $skipRegularLog
    */
-  public function log($message) {
+  public function log($message, $type = NULL, $skipRegularLog = FALSE) {
+    $this->detailedTaskLogs[] = $this->prepareDetailedTaskLogs($message, $type);
+
+    if ($skipRegularLog) {
+      return;
+    }
+
     $message = "[Task {$this->getID()}] {$message}";
     $this->log_messages[] = $message;
     if ($this->log_to_file) {
       CRM_Core_Error::debug_log_message($message, FALSE, 'sqltasks');
     }
   }
 
+  /**
+   * @param $message
+   * @param $type
+   * @return array
+   */
+  private function prepareDetailedTaskLogs($message, $type) {
+    if (empty($type)) {
+      $type = $this->status;
+    }
+
+    $microseconds = microtime(TRUE);
+    $now = DateTime::createFromFormat('U.u', $microseconds);
+
+    return [
```

Comment:
I think we can probably remove some of these items. We run a lot of tasks daily, so reducing size is a concern here. These seem a bit redundant:
- `task_id`
- `date_formatted` (can be derived from timestamp)
- not sure if we need both `task_status`/`message_type`